hollow-jar/full/src/test/java/org/wildfly/swarm/ts/hollow/jar/full/FullHollowJarIT.java

```diff
@@ -0,0 +1,100 @@
+        assertThat(response).isEqualTo("1");
+
+        String resultsURL = "http://localhost:8080/batch?operation=results";
+
+        await().
+            atLeast(10, TimeUnit.MILLISECONDS).
+            atMost(20, TimeUnit.MILLISECONDS).
+        until(() -> {
+            return !"".equals(Request.Get(resultsURL).execute().returnContent().asString()); 
+            });
+
+        response = Request.Get(resultsURL).execute().returnContent().asString();
+
+        assertThat(response).isEqualTo("0\n1\n2\n3\n4\n5\n6\n7\n8\n9\n");
+    }
+
+    @Test
+    public void testMessaging() throws IOException {
+        String response = Request.Get("http://localhost:8080/messaging?operation=sendQueue").execute().returnContent().asString();
+        assertThat(response).isEqualTo("OK");
+
+        String resultsURL = "http://localhost:8080/messaging?operation=results";
+        await().
+            atLeast(10, TimeUnit.MILLISECONDS).
+            atMost(50, TimeUnit.MILLISECONDS).
+        until(() -> {
+          return !"".equals(Request.Get(resultsURL).execute().returnContent().asString()); 
+        });
+
+        response = Request.Get(resultsURL).execute().returnContent().asString();
+        assertThat(response).isEqualTo("1\n");
+
+        response = Request.Get("http://localhost:8080/messaging?operation=sendTopic").execute().returnContent().asString();
+        assertThat(response).isEqualTo("OK");
+
+        await().
+            atLeast(10, TimeUnit.MILLISECONDS).
+            atMost(50, TimeUnit.MILLISECONDS).
+        until(() -> {
+            return !"".equals(Request.Get(resultsURL).execute().returnContent().asString()); 
+        });
+
+        response = Request.Get(resultsURL).execute().returnContent().asString();
+        assertThat(response).isEqualTo("2\n");
+    }
+
+    @Test
+    public void testEjb() throws IOException {
+        String response = Request.Get("http://localhost:8080/remoteejb").execute().returnContent().asString();
+        assertThat(response).isEqualTo("remote ejb method");
+    }
+
+    @Test
+    public void testMail() throws IOException {
+        String response = Request.Get("http://localhost:8080/mail?operation=send").execute().returnContent().asString();
+        assertThat(response).isEqualTo("OK");
+
+        String resultsURL = "http://localhost:8080/mail?operation=results";
+        await().
+            atLeast(10, TimeUnit.MILLISECONDS).
+            atMost(50, TimeUnit.MILLISECONDS).
+        until(() -> {
+          return !"".equals(Request.Get(resultsURL).execute().returnContent().asString()); 
+        });
+
+        response = Request.Get(resultsURL).execute().returnContent().asString();
+        assertThat(response).isEqualTo("FOO\n");
+
+    }
+
+    @Test
+    public void testJaxWS() throws IOException {
+        String response = Request.Get("http://localhost:8080/TestWebService?wsdl").execute().returnContent().asString();
+
+        assertThat(response).isNotBlank();
+    }
+}
```

Comment:
Where's a test for the resource adapter (`net.java.xadisk`)?

Reply:
If it deploys, it works. I think I could get the remote resource adapter resource (as we have two different java -jar processes) but I would need the port etc and make things a bit more complicated, probably without the actual need. In current resource adapter tests it just test it is deploying without errors.

Reply:
As I wrote above, we don't want to have two processes -- with that, we don't really test that one can deploy a resource adapter together with an actual application. Also doing some simple operation with the resource adapter shouldn't be too hard :-)